diff --git a/ijkmedia/ijkplayer/ff_ffplay.c b/ijkmedia/ijkplayer/ff_ffplay.c
index f0366cb2..24f76121 100755
--- a/ijkmedia/ijkplayer/ff_ffplay.c
+++ b/ijkmedia/ijkplayer/ff_ffplay.c
@@ -3052,7 +3052,7 @@ static int read_thread(void *arg)
     FFPlayer *ffp = arg;
     VideoState *is = ffp->is;
     AVFormatContext *ic = NULL;
-    int err, i, ret __unused;
+    int err = 0, i, ret = 0;
     int st_index[AVMEDIA_TYPE_NB];
     AVPacket pkt1, *pkt = &pkt1;
     int64_t stream_start_time;
@@ -3502,7 +3502,8 @@ static int read_thread(void *arg)
                     toggle_pause(ffp, 1);
                     if (ffp->error) {
                         av_log(ffp, AV_LOG_INFO, "ffp_toggle_buffering: error: %d\n", ffp->error);
-                        ffp_notify_msg1(ffp, FFP_MSG_ERROR);
+                        ffp->last_error = ffp->error;
+                        ffp_notify_msg2(ffp, FFP_MSG_ERROR, ffp->error);
                     } else {
                         av_log(ffp, AV_LOG_INFO, "ffp_toggle_buffering: completed: OK\n");
                         ffp_notify_msg1(ffp, FFP_MSG_COMPLETED);
@@ -3624,4 +3625,12 @@ static int read_thread(void *arg)
     ret = 0;
  fail:
+    if (last_error == 0) {
+        if (err < 0)
+            last_error = err;
+        else if (ret < 0)
+            last_error = ret;
+        else
+            last_error = AVERROR_UNKNOWN;
+    }
     if (ic && !is->ic)
         avformat_close_input(&ic);
