diff --git a/libavcodec/aarch64/h264dsp_init_aarch64.c b/libavcodec/aarch64/h264dsp_init_aarch64.c
index e0f378f5ab4..eb2014e24d6 100644
--- a/libavcodec/aarch64/h264dsp_init_aarch64.c
+++ b/libavcodec/aarch64/h264dsp_init_aarch64.c
@@ -34,20 +34,20 @@ void ff_h264_v_loop_filter_chroma_neon(uint8_t *pix, int stride, int alpha,
 void ff_h264_h_loop_filter_chroma_neon(uint8_t *pix, int stride, int alpha,
                                        int beta, int8_t *tc0);

-void ff_weight_h264_pixels_16_neon(uint8_t *dst, int stride, int height,
+void ff_weight_h264_pixels_16_neon(uint8_t *dst, ptrdiff_t stride, int height,
                                    int log2_den, int weight, int offset);
-void ff_weight_h264_pixels_8_neon(uint8_t *dst, int stride, int height,
+void ff_weight_h264_pixels_8_neon(uint8_t *dst, ptrdiff_t stride, int height,
                                   int log2_den, int weight, int offset);
-void ff_weight_h264_pixels_4_neon(uint8_t *dst, int stride, int height,
+void ff_weight_h264_pixels_4_neon(uint8_t *dst, ptrdiff_t stride, int height,
                                   int log2_den, int weight, int offset);

-void ff_biweight_h264_pixels_16_neon(uint8_t *dst, uint8_t *src, int stride,
+void ff_biweight_h264_pixels_16_neon(uint8_t *dst, uint8_t *src, ptrdiff_t stride,
                                      int height, int log2_den, int weightd,
                                      int weights, int offset);
-void ff_biweight_h264_pixels_8_neon(uint8_t *dst, uint8_t *src, int stride,
+void ff_biweight_h264_pixels_8_neon(uint8_t *dst, uint8_t *src, ptrdiff_t stride,
                                     int height, int log2_den, int weightd,
                                     int weights, int offset);
-void ff_biweight_h264_pixels_4_neon(uint8_t *dst, uint8_t *src, int stride,
+void ff_biweight_h264_pixels_4_neon(uint8_t *dst, uint8_t *src, ptrdiff_t stride,
                                     int height, int log2_den, int weightd,
                                     int weights, int offset);

diff --git a/libavutil/dict.c b/libavutil/dict.c
index 48209ac8c73..6f6e4c57207 100644
--- a/libavutil/dict.c
+++ b/libavutil/dict.c
@@ -157,35 +157,34 @@ int av_dict_set_intptr(AVDictionary **pm, const char *key, uintptr_t value,
                 int flags)
 {
     char valuestr[22];
-    snprintf(valuestr, sizeof(valuestr), "%p", value);
+    snprintf(valuestr, sizeof(valuestr), "%p", (void *)value);
     flags &= ~AV_DICT_DONT_STRDUP_VAL;
     return av_dict_set(pm, key, valuestr, flags);
 }

 uintptr_t av_dict_get_intptr(const AVDictionary *m, const char* key) {
-    uintptr_t ptr = NULL;
     AVDictionaryEntry *t = NULL;
     if ((t = av_dict_get(m, key, NULL, 0))) {
       return av_dict_strtoptr(t->value);
     }
-    return NULL;
+    return 0;
 }

 uintptr_t av_dict_strtoptr(char * value) {
-   uintptr_t ptr = NULL;
+   uintptr_t ptr = 0;
    char *next = NULL;
    if(!value || value[0] !='0' || (value[1]|0x20)!='x') {
-       return NULL;
+       return 0;
    }
    ptr = strtoull(value, &next, 16);
    if (next == value) {
-       return NULL;
+       return 0;
    }
    return ptr;
 }

 char * av_dict_ptrtostr(uintptr_t value) {
     char valuestr[22] = {0};
-    snprintf(valuestr, sizeof(valuestr), "%p", value);
+    snprintf(valuestr, sizeof(valuestr), "%p", (void *)value);
     return av_strdup(valuestr);
 }
